### ISSUE TITLE
Photos not always correctly downloaded

### ISSUE THREAD

**Issue by alexanderisme** [ubports/telegram-app#45]
Sometimes when you receive photos there are some that are not correctly downloaded (it mainly happens when receiving a lot of images). (see screen-shot)

I have tested some things: 

- if you download the image and open if with the gallery app it looks the same as on telegram
- if you forward the image in telegram, it will appear entirely (maybe because telegram is cloud based?)

I don't know how this could be solved, maybe a "refresh"? 
![image](https://user-images.githubusercontent.com/29466789/27202334-71429ffa-5221-11e7-9c0c-ea4d80c1bbfd.png)

**Comment by Flohack74** [comment]
Hi, yes the thing is that all pictures are in the cloud and if forwarded, they point again to the same information, which is obviously downloaded again and then ok. I know that the current Telegram has lots of problems, and we try to look into this. Sorry for the experience.

**Comment by alexanderisme** [comment]
Thank you a lot for your work and to continue supporting Telegram and making ubuntu touch continue to live !!!
Yes, there are still some bugs in telegram but it is working really really good!!! 

Is showing the bugs I find as a user a help for development ? Because my aim is not to say that the app is bad, because it's a wonderful app!

**Comment by Flohack74** [comment]
Yes please, we need all feedback that we can get, so don't hesitate and open tickets here ;)

**Comment by corvusd** [comment]
2 Ideas:

1) If the client will can delete the source, when the client should re-download the photo?
2) And another way... the hard way... implement de download in telegram client via download-manager like podbird?

:D

**Comment by Flohack74** [comment]
Telegram uses internally already full asynchronous processing of up and downloads. Its in fact good solution, but probably with lot of bugs ;)

**Comment by corvusd** [comment]
And in this cases, when the archive not download well, not have redownload API option?

**Comment by Flohack74** [comment]
It has but the App markes the download as done. Mor specifically, it just checks if the file exists  :(

**Comment by corvusd** [comment]
And possible delete. Not exist in this device. And reload?

**Comment by Flohack74** [comment]
yes if you would delete it when the download aborts, it would try to re-download. Maybe I can look if this would work

**Comment by Flohack74** [comment]
Actually, this is not just photos, any download can be corrupted.

**Comment by corvusd** [comment]
Yes, it is.
